pycytominer/cyto_utils/load.py

```diff
@@ -69,13 +77,66 @@ def infer_delim(file: str):
     return dialect.delimiter
 
 
-def load_profiles(profiles):
+def is_anndata(
+    path: Union[str, pathlib.Path, ad._core.anndata.AnnData],
+) -> tuple[bool, Optional[str]]:
+    """Return True if ``path`` contains an AnnData dataset (H5AD or Zarr).
+
+    This function prefers using the AnnData readers directly:
+    - in-memory AnnData objects are recognized directly.
+    - H5AD files are opened in backed mode to avoid loading data into memory.
+    - Zarr stores (directories or files like ``.zarr`` or ``.zip``) are read
+      via :func:`anndata.read_zarr`.
+
+    The function is conservative: on any read error (or if AnnData is not
+    installed), it returns ``False``.
+
+    Args:
+        path: File or directory to inspect.
+
+    Returns:
+        Tuple containing:
+        - Bool: True if the path appears to contain an AnnData dataset; otherwise
+        False.
+        - Str: If the path is an AnnData dataset, the type of store
+    """
+
+    # passthrough for anndata in-memory objects
+    if isinstance(path, ad._core.anndata.AnnData):
```

Comment:
it would be great to get confirmation from the `anndata` folks if this covers 100% of anndata objects

Reply:
yes, but under no circumstances should you access private submodules!


```suggestion
    if isinstance(path, ad.AnnData):
```